numba_cuda/numba/cuda/tests/cudapy/test_debuginfo.py

```diff
@@ -784,6 +790,236 @@ def foo(a):
         ir = foo.inspect_llvm()[sig]
         self.assertFileCheckMatches(ir, foo.__doc__)
 
+    def test_missing_source(self):
+        strsrc = """
+        def foo():
+            pass
+        """
+        l = dict()
+        exec(dedent(strsrc), {}, l)
+        foo = cuda.jit(debug=True)(l["foo"])
+
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always", NumbaDebugInfoWarning)
+            ignore_internal_warnings()
+            foo[1, 1]()
+
+        self.assertEqual(len(w), 1)
+        found = w[0]
+        self.assertEqual(found.category, NumbaDebugInfoWarning)
+        msg = str(found.message)
+        # make sure the warning contains the right message
+        self.assertIn("Could not find source for function", msg)
+        # and refers to the offending function
+        self.assertIn(str(foo.py_func), msg)
+
+    def test_irregularly_indented_source(self):
+        @cuda.jit(tuple(), debug=True)
+        def foo():
+            # NOTE: THIS COMMENT MUST START AT COLUMN 0 FOR THIS SAMPLE CODE TO BE VALID # noqa: E115, E501
```

Comment:
This doesn't start at column 0 - I suspect the filecheck tests below might be checking for the wrong thing?

Reply:
You're right, looks like my formatter indented this for me and it still passed so I didn't notice. The DILocation in the IR still has `column: 1`, and I'm not sure why.

Reply:
It looks like the original test was not checking for column zero, just that there was one debug location? I'm not sure how useful of a test it is, but to match the upstream test, we can keep the check for one DIlocation and no others, I think.

Reply:
Test removed from this PR because of issues described above.